Week-4/assignment1.js

```diff
@@ -0,0 +1,57 @@
+//executor function
+function getNumber(resolve, reject) {
+  const randomNum = parseInt(Math.random() * 100);
+  if (randomNum % 5 === 0) {
+    resolve(randomNum); //resolve promise
+  } else {
+    reject(randomNum); //reject promise
+  }
+}
+
+class CustomPromise {
+  constructor(executorFunction) {
+    this.state = "pending";
+    this.value = null;
+
+    //resolve func of promise checks state & updates
+    const resolve = (val) => {
+      if (this.state === "pending") {
+        this.state = "fulfilled";
```

Comment:
if it is pending how can be fulfilled ?

Reply:
Here, i am trying to verify that only the pending promises should be fulfilled, incase "resolve" function is called